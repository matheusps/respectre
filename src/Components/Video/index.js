import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const Video = ({ src, ratio, className, children }) => {
  const videoClasses = classNames(className, 'video-responsive', {
    'video-responsive-16-9': ratio === '16:9',
    'video-responsive-4-3': ratio === '4:3',
    'video-responsive-1-1': ratio === '1:1',
  })
  return !!src ? (
    <video className={videoClasses} src={src} />
  ) : (
    <div className={videoClasses}>{children}</div>
  )
}

Video.propTypes = {
  /** Video Source */
  src: PropTypes.string,
  /** Video aspect ratio */
  ratio: PropTypes.oneOf(['16:9', '4:3', '1:1']),
  /** Aditional css classes */
  className: PropTypes.string,
}

Video.defaultProps = {
  ratio: '16:9',
}

export default Video
