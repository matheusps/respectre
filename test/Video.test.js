import React from 'react'
import Video from '../src/Components/Video'
import renderer from 'react-test-renderer'

describe('Video', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<Video src="testing" />).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should match snapshot with props', () => {
    const ratio169 = renderer
      .create(<Video src="testing" ratio="16:9" />)
      .toJSON()
    const ratio11 = renderer
      .create(<Video src="testing" ratio="1:1" />)
      .toJSON()
    const ratio43 = renderer
      .create(<Video src="testing" ratio="4:3" />)
      .toJSON()

    expect(ratio169).toMatchSnapshot()
    expect(ratio11).toMatchSnapshot()
    expect(ratio43).toMatchSnapshot()
  })

  it('should match snapshot with children', () => {
    const component = renderer
      .create(
        <Video>
          <iframe />
        </Video>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
