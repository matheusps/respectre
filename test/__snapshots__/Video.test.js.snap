// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video should match snapshot 1`] = `
<video
  className="video-responsive video-responsive-16-9"
  src="testing"
/>
`;

exports[`Video should match snapshot with children 1`] = `
<div
  className="video-responsive video-responsive-16-9"
>
  <iframe />
</div>
`;

exports[`Video should match snapshot with props 1`] = `
<video
  className="video-responsive video-responsive-16-9"
  src="testing"
/>
`;

exports[`Video should match snapshot with props 2`] = `
<video
  className="video-responsive video-responsive-1-1"
  src="testing"
/>
`;

exports[`Video should match snapshot with props 3`] = `
<video
  className="video-responsive video-responsive-4-3"
  src="testing"
/>
`;
